prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -110,7 +110,7 @@ def map_lines_to_substations_using_coords(
     }
     for a in all2one:
         type2id = substations.loc[list(a)].reset_index().groupby("TYPE").first()["ID"]
-        for t in ["SUBSTATION", "TAP", "RISER", "DEAD END"]:
+        for t in ["SUBSTATION", "TAP", "RISER", "DEAD END", "NOT AVAILABLE"]:
```

Comment:
So "NOT AVAILABLE" is not only one of the status of substations but also one of substation types that we didn't consider in picking those as a primary substation before?

Reply:
Exactly. Some of the `NOT AVAILABLE` status substations also have type `NOT AVAILABLE`, so we need to add to this list here, otherwise we end up with NaN/None values downstream when the only substation at a given location is not one of `["SUBSTATION", "TAP", "RISER", "DEAD END"]`.